dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/db/sql/PostgreSqlBuilder.java

```diff
@@ -239,52 +241,91 @@ public String createTable(Table table) {
             .append(" (");
 
     // Columns
-
     if (table.hasColumns()) {
-      for (Column column : table.getColumns()) {
-        String dataType = getDataTypeName(column.getDataType());
-        String nullable = column.getNullable() == Nullable.NOT_NULL ? " not null" : " null";
-        String collation = column.getCollation() == Collation.C ? (" collate " + quote("C")) : "";
-
-        sql.append(quote(column.getName()) + " ")
-            .append(dataType)
-            .append(nullable)
-            .append(collation)
-            .append(COMMA);
-      }
+      sql.append(columnsStatement(table.getColumns()));
     }
 
     // Primary key
-
     if (table.hasPrimaryKey()) {
-      sql.append("primary key (");
-
-      for (String columnName : table.getPrimaryKey()) {
-        sql.append(quote(columnName)).append(COMMA);
-      }
-
-      removeLastComma(sql).append(")").append(COMMA);
+      sql.append(primaryKeysStatement(table.getPrimaryKey()));
     }
 
     // Checks
-
     if (table.hasChecks()) {
-      for (String check : table.getChecks()) {
-        sql.append("check(" + check + ")").append(COMMA);
-      }
+      sql.append(checksStatement(table.getChecks()));
     }
 
     removeLastComma(sql).append(")");
 
     // Parent
-
-    if (table.hasParent()) {
+    // Only use partitioned (inherited) tables when we should not distribute the table.
+    if (table instanceof AnalyticsTable analyticsTable
+        && !analyticsTable.isTableDistributed()
+        && table.hasParent()) {
       sql.append(" inherits (").append(quote(table.getParent().getName())).append(")");
     }
 
     return sql.append(";").toString();
   }
 
+  /**
+   * Returns the SQL statement for the given list of checks.
+   *
+   * @param checks the list of checks to append to the SQL statement.
+   * @return the SQL statement.
+   */
+  private static String checksStatement(List<String> checks) {
+    StringBuilder sql = new StringBuilder();
+
+    for (String check : checks) {
+      sql.append("check(" + check + ")").append(COMMA);
+    }
+
+    return sql.toString();
+  }
+
+  /**
+   * Returns the SQL statement for the given list of primaryKeys.
+   *
+   * @param primaryKeys the list of primaryKeys to append to the SQL statement.
+   * @return the SQL statement.
+   */
+  private String primaryKeysStatement(List<String> primaryKeys) {
```

Comment:
Do we need to split this into separate methods? I know Sonar complains but I think it is more readable in one method/the way it was.

Reply:
Splitting them into separate, as suggested by Sonar made more sense. But I will revert as you wish. No problems.